src/app/core/services/validation.service.ts

```diff
@@ -62,62 +62,37 @@ export class ValidationService {
 
 	async refineTimebound(dataset: Dataset, property: Property): Promise<ValidationResponse | null> {
 		const pattern = property.getPattern();
-		if (pattern instanceof Response) {
+		if (pattern instanceof Response || pattern instanceof Absence || pattern instanceof Universality) {
 			let start = 0;
 			let end = dataset.data.length - 1;
 			const validationResponses: ValidationResponse[] = [];
 			while (start <= end) {
-
 				let mid = Math.floor((start + end) / 2);
 				const propertyCandidate = Object.assign(Object.create(Object.getPrototypeOf(property)), property);
 				const patternCandidate = Object.assign(Object.create(Object.getPrototypeOf(pattern)), pattern);
-				patternCandidate.setSTimeBound(new Interval(new Event('Candidate' + end), 0, end, 'time units'));
-				propertyCandidate.setPattern(patternCandidate);
-				const validationResponse = await this.validateProperty(dataset, propertyCandidate);
-				validationResponses.push(validationResponse);
-				if (validationResponse.result === true) {
-					end = mid - 1;
-				} else {
-					start = mid + 1;
-				}
 
-
-				/* let mid = Math.floor((start + end) / 2);
-				const propertyCandidate = Object.assign(Object.create(Object.getPrototypeOf(property)), property);
-				const patternCandidate = Object.assign(Object.create(Object.getPrototypeOf(pattern)), pattern);
-				patternCandidate.setSTimeBound(new Interval(new Event('Candidate' + mid), 0, mid, 'time units'));
+				if (pattern instanceof Response) {
+					patternCandidate.setSTimeBound(new Interval(new Event('Candidate' + end), 0, end, 'time units'));
+				} else if (pattern instanceof Absence || pattern instanceof Universality) {
```

Comment:
Could be problematic later to not check the scope - currently, there is just one scope for each pattern, so it is clear that only the After-Scope is set for Absence and Universality, but this might break as soon as more scopes become available.